File: Assignment_21/program2.c
// Accept N numbers from user and return the smallest number.

///////////////////////////////////////////////////////////////////////////////
// File Name   :  program2.c
// Description :  Accept N numbers from user and return the smallest number
// Autor       :  Ankita Anil Patil
// Date        :  31/05/2025
///////////////////////////////////////////////////////////////////////////////


/*
 Input : N : 6
         Elements : 85 66 3 66 93 88
 Output: 3

 */

 #include<stdio.h>
 #include<stdlib.h>
 
  int Minimum(int Arr[], int iSize)
  {
     if((Arr == NULL ) || (iSize <= 0))
     {
         printf("Invalid input ");
         return -1;
     }
 
     int i = 0, iMini = 0;
 
     iMini = Arr[0];
     for(i = 0; i < iSize; i++)
     {
         if(Arr[i] < iMini)
         {
                iMini = Arr[i];
         }
     }
 
     return iMini;
  }
 
  int main()
  {
     int iLength = 0, iCnt = 0, iRet = 0;
     int *ptr = NULL;
 
     printf("Enter number of elements \n");
     scanf("%d",&iLength);
 
     ptr = (int *) malloc(iLength * sizeof(int));
 
     if(NULL == ptr)
     {
         printf("Unable to allocate memory");
         return -1;
     }
 
     printf("Enter elements\n");
 
     for(iCnt = 0; iCnt < iLength; iCnt++)
     {
         scanf("%d",&ptr[iCnt]);
     }
 
     iRet = Minimum(ptr,iLength);
     printf("Minimum number is %d\n",iRet);
 
     return 0;
  }

File: Assignment_45/program2.java
// Write a program which accept matrix and reverse the contents of each row.

/* 
Input :

      3 2 5 9
      4 3 2 2
      8 4 1 9
      3 9 7 5
Output :

      9 5 2 3
      2 2 3 4
      9 1 4 8
      5 7 9 3
*/


///////////////////////////////////////////////////////////////////////////////
// File Name   :  program2.java
// Description :  Accept matrix and reverse the contents of each row
// Autor       :  Ankita Anil Patil
// Date        :  2/08/2025
///////////////////////////////////////////////////////////////////////////////


import java.util.*;
class Matrix
{
    private  int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int[A][B];
        
    }
    
    public void Accept()
    {
        System.out.println("Please enter elements of matrix ");
      
        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0 ; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
            
            System.out.println();
        }
     
    }

     public void Display()
     {
         System.out.println("Elemnets of Matrix are : ");
         int i = 0, j = 0;
         
         for(i = 0; i < Arr.length; i++)
          {
               for(j = 0 ; j < Arr[i].length; j++)
               {
                     System.out.print(Arr[i][j]+"\t");
               }
               System.out.println();
            
          }
     }


    public void ReverseRow() 
    { 
          
          int i = 0, j = 0,iStart = 0, iEnd = 0, temp = 0;

          for(i = 0; i < Arr.length  ; i++)
          { 
                iStart = 0;
                iEnd = Arr[i].length -1;
                while (iStart < iEnd) 
                {
                        temp = Arr[i][iStart];
                        Arr[i][iStart] = Arr[i][iEnd];
                        Arr[i][iEnd] = temp;

                        iEnd--;
                        iStart++;
                 
                }

               
            }
    }
 
   
}
     



class program2
{
    public static void main(String A[]) 
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number of rows ");
        int iRow = sobj.nextInt();

        System.out.println("Enter Number of columns ");
        int iCol = sobj .nextInt();

        Matrix mobj = new Matrix(iRow,iCol);
        mobj.Accept();
        mobj.Display();
        mobj.ReverseRow();
        mobj.Display();

        

    }
}